src/main/java/org/folio/processing/matching/reader/util/MatchIdProcessorUtil.java

```diff
@@ -0,0 +1,63 @@
+package org.folio.processing.matching.reader.util;
+
+import org.folio.DataImportEventPayload;
+import org.folio.processing.value.MissingValue;
+import org.folio.processing.value.StringValue;
+import org.folio.processing.value.Value;
+
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+/**
+ * Util class for processing and retrieving ids from context`s data.
+ */
+public final class MatchIdProcessorUtil {
+
+  private static final String MAPPING_PARAMS = "MAPPING_PARAMS";
+  private static final String RELATIONS = "MATCHING_PARAMETERS_RELATIONS";
+  private static final String NAME_PROPERTY = "name";
+  private static final String ID_PROPERTY = "id";
+  private static final String CODE_PROPERTY = "code";
+  private static final String LOCATIONS_PROPERTY = "locations";
+
+  private MatchIdProcessorUtil() {
+  }
+
+  public static Value retrieveIdFromContext(String field, DataImportEventPayload eventPayload, Value value) {
+    JsonObject matchingParams = new JsonObject(eventPayload.getContext().get(MAPPING_PARAMS));
+    JsonObject relations = new JsonObject(eventPayload.getContext().get(RELATIONS));
+    String relation = String.valueOf(relations.getJsonObject("matchingRelations")
```

Comment:
can relations and/or matchingParams be null?

Reply:
@KaterynaSenchenko I think no because they put in the mod-inventory in trivial logic whenever matchHandler triggered. 

Moreover, I will remove "matchingRelations" property from context, and there will be the only Map with key-value, regarding the discussion with @RuslanLavrov.